pkg/controller/statusmanager/status_manager.go

```diff
@@ -65,30 +68,94 @@ func (c ClusteredName) String() string {
 	return c.ClusterName + string(ClusteredNameSeparator) + c.Namespace + string(ClusteredNameSeparator) + c.Name
 }
 
+const generateStatusSelector = names.GenerateStatusLabel // Just a single label is an EXISTS selector. Easy.
+
 // StatusManager coordinates changes to ClusterOperator.Status
 type StatusManager struct {
 	sync.Mutex
 
-	client cnoclient.Client
-	name   string
+	client           cnoclient.Client
+	name             string
+	hyperShiftConfig *network.HyperShiftConfig
 
 	failing         [maxStatusLevel]*operv1.OperatorCondition
 	installComplete bool
 
-	daemonSets     []ClusteredName
-	deployments    []ClusteredName
-	statefulSets   []ClusteredName
-	relatedObjects []configv1.ObjectReference
+	dsInformer cache.SharedIndexInformer
+	dsLister   DaemonSetLister
 
-	hyperShiftConfig *network.HyperShiftConfig
+	depInformer cache.SharedIndexInformer
+	depLister   DeploymentLister
+
+	// StatefulSets may exist in more than one cluster (in the case of Hypershift)
```

Comment:
It seems weird to treat this like there's some inherent difference between StatefulSets and Deployments/DaemonSets, as opposed to this just being a contingent truth about OCP 4.11. (Which is what it is, right?) Maybe there would be some simplifications if you made the three types more consistent in the reorg?

Reply:
This is all a good point. As it stands right now, only StatefulSets are deployed in multiple clusters. That could certainly change, however, so, you're right, this should all take the same shape.